packages/tendermint-rpc/src/rpcclients/httpbatchclient.ts

```diff
@@ -55,6 +60,11 @@ export class HttpBatchClient implements RpcClient {
   public async execute(request: JsonRpcRequest): Promise<JsonRpcSuccessResponse> {
     return new Promise((resolve, reject) => {
       this.queue.push({ request, resolve, reject });
+
+      if (this.queue.length >= this.options.batchSizeLimit) {
+        // this train is full, let's go
+        this.tick();
```

Comment:
Should the timer be reset in this instance?

Reply:
I was thinking about this as well but ended up going with the simplest implementation. I guess it does not matter much.

Before this PR we had a tick after: 20ms, 40ms, 60ms, 80ms, …
With this, assuming 50 initial requests after 12ms of page load: 12ms, 12ms, 20ms, 40ms, 60ms, 80ms, …

So in case of high traffic, we get faster execution requests. The `dispatchInterval` remains the max time a request is delayed to wait for more requests but it is not a constant time.